cmake/libprotobuf-lite.cmake

```diff
@@ -108,6 +108,11 @@ if(MSVC AND protobuf_BUILD_SHARED_LIBS)
     PUBLIC  PROTOBUF_USE_DLLS
     PRIVATE LIBPROTOBUF_EXPORTS)
 endif()
+if(${CMAKE_SYSTEM_NAME} STREQUAL "iOS" AND protobuf_BUILD_SHARED_LIBS)
```

Comment:
I think we should simplify this by just enabling this behavior regardless of platform (but still conditional on `protobuf_BUILD_SHARED_LIBS`. So the MSVC and iOS blocks could be merged into one:
```
if(protobuf_BUILD_SHARED_LIBS)
  target_compile_definitions(libprotobuf-lite
    PUBLIC  PROTOBUF_USE_DLLS
    PRIVATE LIBPROTOBUF_EXPORTS)
endif()
```
And same thing in the other file.

Reply:
That make sense!